src/components/banner/Banner.js

```diff
@@ -0,0 +1,63 @@
+import MockBanner from "../../mocks/en-us/featured-banners.json"
+import React, { useRef, useEffect } from 'react';
+import * as styles from './Banner-style';
+
+export default function Banner(){
+    const banner = MockBanner
+    
+
+    const slideref = useRef(null);
+
+    const Slider = () => {
+        if(slideref.current.children.length > 0){
+            const firstElement = slideref.current.children[0];
+
+            slideref.current.style.transition = `700ms ease-in all`;
+            slideref.current.style.transform = `translateX(-100%)`;
+            const append = () => {
+                slideref.current.style.transition = 'none';
+                slideref.current.style.transform = `translateX(0)`;
+                slideref.current.appendChild(firstElement);
+                slideref.current.removeEventListener('transitionend', append);
+            }
+            slideref.current.addEventListener('transitionend', append);
+        } 
+    }
+
+    // useEffect(()=>{
```

Comment:
Its a best practice to put comments to help other devs to understand these lines, or delete them